packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -90,6 +116,174 @@ private static BitmapDescriptor getBitmapFromBytes(List<?> data) {
     }
   }
 
+  /**
+   * Creates a BitmapDescriptor object from bytes data.
+   *
+   * <p>This method requires the `byteData` map to contain specific keys: 'byteData' for image
+   * bytes, 'bitmapScaling' for scaling mode, and 'imagePixelRatio' for scale ratio. It may
+   * optionally include 'width' and/or 'height' for explicit image dimensions.
+   *
+   * @param byteData a map containing the byte data and scaling instructions. Expected keys are:
+   *     'byteData': the actual bytes of the image, 'bitmapScaling': the scaling mode, either 'auto'
+   *     or 'none', 'imagePixelRatio': used with 'auto' bitmapScaling if width or height are not
+   *     provided, 'width' (optional): the desired width, which affects scaling if 'height' is not
+   *     provided, 'height' (optional): the desired height, which affects scaling if 'width' is not
+   *     provided
+   * @param density the density of the display, used to calculate pixel dimensions.
+   * @param bitmapDescriptorFactory is an instance of the BitmapDescriptorFactoryWrapper.
+   * @return BitmapDescriptor object from bytes data.
+   * @throws IllegalArgumentException if any required keys are missing in `byteData` or if the byte
+   *     data cannot be interpreted as a valid image.
+   */
+  @VisibleForTesting
+  public static BitmapDescriptor getBitmapFromBytes(
+      Map<?, ?> byteData, float density, BitmapDescriptorFactoryWrapper bitmapDescriptorFactory) {
+
+    if (!byteData.containsKey("byteData")) {
```

Comment:
nit: Consider making "byteData", "bitmapScaling" and "imagePixelRatio" consts in this method.